src/contrib/tevent.rst

```diff
@@ -0,0 +1,722 @@
+            }
+
+            int main()
+            {
+                struct tevent_context *ev;
+                struct tevent_immediate *im;
+                int i;
+
+                /* Initialize tevent context. */
+                ev = tevent_context_init(NULL);
+                if (ev == NULL) {
+                    return 1;
+                }
+
+                im = tevent_create_immediate(ev);
+                if (im == NULL) {
+                    return 2;
+                }
+
+                tevent_schedule_immediate(im, ev, handler, NULL);
+                tevent_loop_wait(ev);
+
+                return 0;
+            }
+
+        .. code-block:: console
+
+            $ gcc ./main.c -ltalloc -ltevent
+            $ ./a.out
+            Hello world.
+
+    .. plain-tab::
+        :label: Signal example
+
+        .. code-block:: c
+
+            /**
+            * React on SIGUSR1 signal.
+            */
+
+            #include <signal.h>
+            #include <stdio.h>
+            #include <tevent.h>
+
+            static void handler(struct tevent_context *ev,
+                                struct tevent_signal *se,
+                                int signum,
+                                int count,
+                                void *siginfo,
+                                void *private_data)
+            {
+                puts("Signal received.");
+            }
+
+            int main()
+            {
+                struct tevent_context *ev;
+                struct tevent_signal *sig;
+                int i;
+
+                /* Initialize tevent context. */
+                ev = tevent_context_init(NULL);
+                if (ev == NULL) {
+                    return 1;
+                }
+
+                sig = tevent_add_signal(ev, NULL, SIGUSR1, 0, handler, NULL);
+                if (sig == NULL) {
+                    return 2;
+                }
+
+                tevent_loop_wait(ev);
+
+                return 0;
+            }
+
+        .. code-block:: console
+
+            $  $ gcc ./main.c -ltalloc -ltevent             |
+            $ ./a.out                                       |
+                                                            |   $ kill -s SIGUSR1 $(pidof a.out)
+            Signal received.                                |
+                                                            |   $ kill -s SIGUSR1 $(pidof a.out)
+            Signal received.                                |
+
+.. note::
+
+    It is important to understand how the event works, however you most likely
+    will not need to work with them directly since SSSD already provides a
+    higher level API around those that is implemented via tevent request.
+
+    :tag:`strong` `Tevent requests`_ are fundamental part of SSSD code and you will work with
+    them and create them all the time, so it is recommended to pay extra
+    attention to the following chapter. :end-tag:`strong`
```

Comment:
These two `note` sentences appear to contradict each other. Is the first sentence referring to non-request event types? If so, it could help to make that more obvious.

Reply:
Yes, to non-request event types. How about now?